src/jit/codegencommon.cpp

```diff
@@ -5340,7 +5340,15 @@ void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pIni
 #endif // !_TARGET_XARCH_
 
 #if CPU_LOAD_STORE_ARCH || !defined(_TARGET_UNIX_)
+        if (initReg == REG_FPBASE)
+        {
+            this->regSet.rsSetRegsModified(RBM_FPBASE);
+        }
         instGen_Set_Reg_To_Zero(EA_PTRSIZE, initReg);
+        if (initReg == REG_FPBASE)
+        {
+            this->regSet.rsRemoveRegsModified(RBM_FPBASE);
+        }
```

Comment:
@cshung - I'm going to have to take a closer look at this. I gather that `initReg` is a free register allocated by the code generator for this purpose, but not by the register allocator (hence it may not be marked modified). However, (as I understand it), if you don't un-mark it after, you run into an assert. But I would imagine that the assert would only fire if we're actually using `REG_FPBASE` as a frame pointer - and in that case, it would seem to me to be incorrect to use it as the `initReg`.

Reply:
@CarolEidt It has been a few days, did you figure out what is going on here?

When I change the implementation of verifyRegUsed() from rsSetRegsModified() to actually asserting the register are already marked as modified, there are a few more scenarios where it asserts the register were not already marked. Do we want to investigate those as well?